File: Data Structures & Algorithms/same-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean compareDFS(TreeNode p, TreeNode q) {
        if (p == null && q == null) {
            return true;
        }
        if (p == null || q == null) {
            return false;
        }

        boolean left = compareDFS(p.left, q.left);
        boolean right = compareDFS(p.right, q.right);

        return p.val == q.val && left && right;
    }

    public boolean isSameTree(TreeNode p, TreeNode q) {
        return compareDFS(p, q);
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []

        res = []
        q = [root]

        while q: # a cada andar
            res.append([])
            aux = []
            while q:
                node = q.pop(0)
                aux.append(node)
                res[-1].append(node.val)
            
            for node in aux:
                for side in [node.left, node.right]:
                    if side:
                        q.append(side)

        return res
